pithos/pithos.py

```diff
@@ -1333,20 +1351,34 @@ def on_menuitem_create_song_station(self, widget):
     def on_treeview_button_press_event(self, treeview, event):
         x = int(event.x)
         y = int(event.y)
+        time = event.time
         pthinfo = treeview.get_path_at_pos(x, y)
         if pthinfo is not None:
             path, col, cellx, celly = pthinfo
             treeview.grab_focus()
             treeview.set_cursor( path, col, 0)
 
             if event.button == 3:
-                rating = self.selected_song().rating
-                self.song_menu_love.set_property("visible", rating != RATE_LOVE)
-                self.song_menu_unlove.set_property("visible", rating == RATE_LOVE)
-                self.song_menu_ban.set_property("visible", rating != RATE_BAN)
-                self.song_menu_unban.set_property("visible", rating == RATE_BAN)
-
-                popup_at_pointer(self.song_menu, event)
+                if not self.selected_song().is_ad:
+                    rating = self.selected_song().rating
+                    self.song_menu_love.set_property("visible", rating != RATE_LOVE)
+                    self.song_menu_unlove.set_property("visible", rating == RATE_LOVE)
+                    self.song_menu_ban.set_property("visible", rating != RATE_BAN)
+                    self.song_menu_unban.set_property("visible", rating == RATE_BAN)
```

Comment:
__[flake8]__
* [F405] 'RATE_BAN' may be undefined, or defined from star imports: .pandora, .pandora.data

<!-- Comment by [SideCI](https://sideci.com) -->